src/integrationtests/java/com/aws/greengrass/integrationtests/status/ComponentStatusChangeFleetStatusServiceTest.java

```diff
@@ -0,0 +1,202 @@
+                FleetStatusDetails fleetStatusDetails = OBJECT_MAPPER.readValue(publishRequest.getPayload(), FleetStatusDetails.class);
+                if (fleetStatusDetails.getTrigger().equals(Trigger.COMPONENT_STATUS_CHANGE)) {
+                    statusChange.countDown();
+                }
+                fleetStatusDetailsList.get().add(fleetStatusDetails);
+            } catch (JsonMappingException ignored) {
+            }
+            return CompletableFuture.completedFuture(0);
+        });
+    }
+    @AfterEach
+    void afterEach() {
+        kernel.shutdown();
+    }
+    @Test
+    void GIVEN_one_component_errored_and_recovered_THEN_fss_should_send_recovery_message() throws Exception {
+        statusChange = new CountDownLatch(2);
+        deviceConfiguration = new DeviceConfiguration(kernel, "ThingName", "xxxxxx-ats.iot.us-east-1.amazonaws.com",
+                "xxxxxx.credentials.iot.us-east-1.amazonaws.com", "privKeyFilePath", "certFilePath", "caFilePath",
+                "us-east-1", "roleAliasName");
+        kernel.getContext().put(DeviceConfiguration.class, deviceConfiguration);
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel,
+                ComponentStatusChangeFleetStatusServiceTest.class.getResource("fss_service_recovery.yaml"));
+        kernel.launch();
+
+        FleetStatusService fss = (FleetStatusService) kernel.locateIgnoreError(FLEET_STATUS_SERVICE_TOPICS);
+        fss.setWaitBetweenPublishDisabled(true);
+
+        assertTrue(statusChange.await(20, TimeUnit.SECONDS));
+
+        // we expect a total of 3 messages; 1 nucleus launch and 2 component status change (error + recovery)
+        assertEquals(3, fleetStatusDetailsList.get().size());
+
+        // the first message should be nucleus launch
+        FleetStatusDetails fleetStatusDetails = fleetStatusDetailsList.get().get(0);
+        assertEquals(OverallStatus.HEALTHY, fleetStatusDetails.getOverallStatus());
+        assertEquals(Trigger.NUCLEUS_LAUNCH, fleetStatusDetails.getTrigger());
+
+        // the second message should be errored component
+        fleetStatusDetails = fleetStatusDetailsList.get().get(1);
+        assertEquals(Trigger.COMPONENT_STATUS_CHANGE, fleetStatusDetails.getTrigger());
+        assertTrue(fleetStatusDetails.getComponentDetails().stream()
+                .anyMatch(c -> "ServiceA".equals(c.getComponentName()) && State.ERRORED.equals(c.getState())));
+
+        // the third message should be component recovery
+        fleetStatusDetails = fleetStatusDetailsList.get().get(2);
+        assertEquals(Trigger.COMPONENT_STATUS_CHANGE, fleetStatusDetails.getTrigger());
+        assertTrue(fleetStatusDetails.getComponentDetails().stream()
+                .anyMatch(c -> "ServiceA".equals(c.getComponentName()) && State.RUNNING.equals(c.getState())));
+    }
+
+    @Test
+    void GIVEN_two_components_errored_and_recovered_THEN_fss_should_send_only_one_recovery_message() throws Exception {
```

Comment:
I don't think this scenario is needed?

Reply:
I think we should still keep it since it's the basic requirement